Sources/Defaults/Defaults+Bridge.swift

```diff
@@ -0,0 +1,258 @@
+import Foundation
+#if os(macOS)
+import AppKit
+#else
+import UIKit
+#endif
+
+extension Defaults.CodableBridge {
+	public func serialize(_ value: Value?) -> Serializable? {
+		guard let value = value else {
+			return nil
+		}
+
+		do {
+			// Some codable values like URL and enum are encoded as a top-level
+			// string which JSON can't handle, so we need to wrap it in an array
+			// We need this: https://forums.swift.org/t/allowing-top-level-fragments-in-jsondecoder/11750
+			let data = try JSONEncoder().encode([value])
+			return String(String(data: data, encoding: .utf8)!.dropFirst().dropLast())
+		} catch {
+			print(error)
+			return nil
+		}
+	}
+
+	public func deserialize(_ object: Serializable?) -> Value? {
+		guard let jsonString = object else {
+			return nil
+		}
+
+		return [Value].init(jsonString: "[\(jsonString)]")?.first
+	}
+}
+
+extension Defaults {
+	public struct TopLevelCodableBridge<Value: Codable>: CodableBridge {}
```

Comment:
Needs a doc comment.

Reply:
Can you split out all of these into separate extensions instead. To reduce nesting.

```swift
extension Defaults {
	public struct TopLevelCodableBridge<Value: Codable>: CodableBridge {}
}
```